File: plugins/spring-boot/src/main/resources/templates/java/spring-boot/webmvc.g8/src/main/g8/server/src/$if(simple.truthy)$.$else$main$endif$/$if(simple.truthy)$.$else$java$endif$/$package$/server/note/NoteDto.java
package $package$.server.note;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import java.time.LocalDateTime;

public final class NoteDto {

    private NoteDto() {}

    public record View(long id, String body, LocalDateTime createdAt) {}

    public record Create(@NotBlank @Size(max = 2000) String body) {}
}

File: plugins/spring-boot/src/main/resources/templates/kotlin/spring-boot/webmvc.g8/src/main/g8/server/src/$if(simple.truthy)$.$else$main$endif$/$if(simple.truthy)$.$else$kotlin$endif$/$package$/server/note/NoteDto.kt
package $package$.server.note

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size
import java.time.LocalDateTime

data class NoteView(val id: Long, val body: String, val createdAt: LocalDateTime)

data class NewNote(
    @field:NotBlank @field:Size(max = 2000) val body: String,
)
